test/test_torch.py

```diff
@@ -10632,6 +10632,17 @@ class MyTwoTensor4(TwoTensor):
         self._checked_swap(t6, t7)
 
 
+    def test_histc_when_min_is_equal_to_max(self):
+        # both min and max are non-zero
+        t = torch.tensor([1, 2, 3, 3, 4, 5, 6, 7, 8], dtype=torch.float)
+        res = torch.histc(t, 5, 7, 7)
+        self.assertEqual(res, torch.tensor([0., 0., 1., 0., 0.]))
```

Comment:
I have read the issue and I must admit to being a bit puzzled by the chosen behavior here. In general the semantics when min/max are equal is a bit special, because there's nothing to put in the bins. It's especially confusing when you have multiple bins: even if you are counting exactly how many elements are == min == max, which bin to put it in? It does not seem well specified.

So... I don't really see why we should make this BC breaking change. The new behavior is not better. Just let sleeping dogs lie (or modify the docs to save that behavior with min == max is undefined when bins > 1).

Reply:
Thanks for your review! This is indeed a special case but there is a difference between PyTorch and NumPy in this case:

```python
if __name__ == '__main__':
    import numpy as np
    data = np.array([1, 2, 3, 3, 4, 5, 6, 7, 8])
    hist, bin_edges = np.histogram(data, bins=5, range=(7, 7))
    print("Histogram:", hist)
    print("Bin edges:", bin_edges)

    import torch
    t = torch.tensor([1, 2, 3, 3, 4, 5, 6, 7, 8], dtype=torch.float)
    res = torch.histc(t, 5, 7, 7)
    print(res)
```

```
Histogram: [0 0 1 0 0]  # Numpy
Bin edges: [6.5 6.7 6.9 7.1 7.3 7.5]
tensor([2., 2., 2., 1., 2.])  # PyTorch
```

I checked the difference in their impelmentations and found that the range would be subtracted by 0.5 instead of 1 on NumPy. See: https://github.com/numpy/numpy/blob/4df5c691744af8f2cfe95c6cbb471ef27f1c22b4/numpy/lib/_histograms_impl.py#L326-L329

Reply:
@lezcano do you think you could find someone to take a closer look here?

Reply:
>In general the semantics when min/max are equal is a bit special, because there's nothing to put in the bins.

This is not really true, as the range is inclusive on both ends. Following this, the current PyTorch behaviour is wrong. I think at least, you want to have `torch.histc(t, b, a, a).sum() == (t == a).sum()` whenever `b > 0`.

Now, as for the distribution of the bins, if `min == max` and not equal to zero and bins > 1, we can either.
- Leave the behaviour unspecified
- Raise an error

I think that leaving the behaviour unspecified is less noisy. Even more, since we can follow NumPy's behaviour without a perf hit, I think this is the way to go.

Reply:
Actually, the proposed fix still chooses "large buckets", so I don't think that will work if the inputs are closer together. See
```python
import numpy as np
data = np.array([6.9, 7., 7.1])
hist, bin_edges = np.histogram(data, bins=5, range=(7, 7))
print(hist)
>[0 0 2 1 0]
```
cc @rgommers for a potential bug in an edgecase in NumPy

Reply:
@rgommers Thanks so much for your reply! We should be careful about this breaking change.